core/templates/dev/head/pages/exploration_player/HintManagerService.js

```diff
@@ -43,6 +46,15 @@ oppia.factory('HintManagerService', [
         }, WAIT_FOR_HINT_MSEC);
         return currentHint.getHtml();
       },
+      getHintAtIndex: function(index) {
+        if (index === numHintsConsumed) {
```

Comment:
I think this should just be a getter and it should perhaps throw an exception if the given index isn't valid. But it seems wrong for it to cause a state change. GETs are supposed to leave things as they are.